### PR TITLE
android: persistent settings

### PR DESCRIPTION
**Pull request by elig0n**
This adds saving & restoring for the settings: subtitles toggle state, IP address, port & password. The saved states are read on startup and are saved everytime they are modified and applied/lose focus.

For the IP if no IP address is already saved (i.e. at first run) or is set to an empty string or app storage was cleared then the usual procedure of default IP prefix construction and scan is done. This still allows people to utilize them on first run.

#34

### CODE REVIEW

File: MpvRemote/app/src/main/java/miccah/laptopremote/MainActivity.java [hunk 1]
```diff
@@ -39,6 +40,11 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        BackgroundToggleButton subtitles2 = (miccah.mpvremote.BackgroundToggleButton)
+                findViewById(R.id.subtitles2);
+        subtitles2.setChecked(PreferenceManager.getDefaultSharedPreferences(this)
+                .getBoolean("SUBTITLES", true));
+
```
Comment (mcastorina) [inline MpvRemote/app/src/main/java/miccah/laptopremote/MainActivity.java:47]: What is this code doing? Where is the `subtitles2` view defined?

Comment (elig0n) [reply MpvRemote/app/src/main/java/miccah/laptopremote/MainActivity.java]: sorry it's a typo.

Comment (mcastorina) [reply MpvRemote/app/src/main/java/miccah/laptopremote/MainActivity.java]: No worries, I see it's getting the `SUBTITLES` preference here but I'm not seeing where it gets set.

Comment (elig0n) [reply MpvRemote/app/src/main/java/miccah/laptopremote/MainActivity.java]: my bad, I got it into another branch by mistake

**Comment by mcastorina** [comment]
Great work!

**Pull request merged**